src/stp/Play.cpp

```diff
@@ -16,51 +16,55 @@ void Play::updateWorld(world_new::World* world) noexcept {
     this->field = world->getField().value();
 }
 
-Status Play::update() noexcept {
-    constexpr static size_t ENUM_COUNT = 4;
-    std::array<size_t, ENUM_COUNT> count{};
-    std::fill(count.begin(), count.end(), 0);
+void Play::update() noexcept {
+    roleStatuses.clear();
```

Comment:
maybe a comment as it is not clear why this is done

Reply:
```suggestion
    // clear roleStatuses so it only contains the current tick's statuses
    roleStatuses.clear();
```